programs/pluto/kernel_xfrm.c

```diff
@@ -387,6 +400,36 @@ static void kernel_xfrm_flush(struct logger *logger)
 			  &recv_errno, logger);
 }
 
+static void print_ext_ack(const struct nlmsghdr *n, struct jambuf *buf)
+{
+	if (n->nlmsg_type != NLMSG_ERROR ||
+	    !(n->nlmsg_flags & NLM_F_ACK_TLVS)) {
+		return;
+	}
+
+	struct nlmsgerr *err = (void *)n + NLMSG_HDRLEN;
+	size_t offset = sizeof(*err);
+	if (!(n->nlmsg_flags & NLM_F_CAPPED)) {
+		offset += err->msg.nlmsg_len - NLMSG_HDRLEN;
+	}
+
+	bool first = true;
+	for (const struct nlattr *attr = nl_getattr(n, &offset);
+	     attr != NULL; attr = nl_getattr(n, &offset)) {
+		if ((attr->nla_type & NLA_TYPE_MASK) == NLMSGERR_ATTR_MSG) {
+			const char *msg = nl_getattrvalstrz(n, attr);
+			if (msg) {
+				if (first) {
+					jam(buf, "netlink ext_ack: %s", msg);
+					first = false;
+				} else {
+					jam(buf, "; %s", msg);
+				}
```

Comment:
If it makes your life easier, you could log one message per log line.  Otherwise, you can `break` out of the `LLOG_JAMBUF()`

Reply:
Done.